orderer/consensus/smartbft/consenter.go

```diff
@@ -261,12 +261,46 @@ func (c *Consenter) IsChannelMember(joinBlock *cb.Block) (bool, error) {
 		return false, errors.New("no orderer config in bundle")
 	}
 	member := false
+
+	santizedCert, err := crypto.SanitizeX509Cert(c.Identity)
+	if err != nil {
+		return false, err
+	}
+
+	// Extract public key using the same approach as IsConsenterOfChannel
+	bl, _ := pem.Decode(santizedCert)
+	if bl == nil {
+		return false, errors.Errorf("node identity certificate %s is not a valid PEM", string(santizedCert))
+	}
+
+	myPublicKey, err := cluster.ExtractPublicKeyFromCert(bl.Bytes)
+	if err != nil {
+		c.Logger.Warningf("Failed to extract public key from own certificate: %v", err)
+		return false, err
+	}
+
 	for _, consenter := range oc.Consenters() {
 		santizedCert, err := crypto.SanitizeX509Cert(consenter.Identity)
 		if err != nil {
+			c.Logger.Warnf("Failed to sanitize consenter %d identity: %v", consenter.Id, err)
 			return false, err
 		}
-		if bytes.Equal(c.Identity, santizedCert) {
+
+		// Extract public key using the same approach as IsConsenterOfChannel
+		bl, _ := pem.Decode(santizedCert)
+		if bl == nil {
+			c.Logger.Warnf("Consenter %d: failed to decode PEM for identity", consenter.Id)
+			continue
+		}
+
+		publicKey, err := cluster.ExtractPublicKeyFromCert(bl.Bytes)
+		if err != nil {
+			c.Logger.Warnf("Consenter %d: failed to extract public key from cert: %v", consenter.Id, err)
+			continue
+		}
+		c.Logger.Debugf("Consenter %d: extracted public key: %x", consenter.Id, publicKey)
+		if bytes.Equal(myPublicKey, publicKey) {
+			c.Logger.Debugf("Found matching public key for consenter %d", consenter.Id)
```

Comment:
How is this change related to what is stated in the problem statement?
This means we ignore enrollment certs as identity only care about the public key. In the description it says the problem is the renewal of TLS certs, which, in BFT, are not part of node identity. 

The code here is a significant change to how fabric orderer BFT handles consenters identity, isn't it?

Reply:
The problem statement is:

https://github.com/hyperledger/fabric/pull/5268#:~:text=Currently%2C%20when%20TLS,maintaining%20backward%20compatibility.

IE: "when TLS certs are renewed". There is nothing there about enrollment certs.